### PR TITLE
fix any_login for production, fix mailer

### PR DESCRIPTION
**Pull request by vikdotdev**
- [x] merged with the latest master branch
- [x] resolved all the conflicts, PR could be merged
- [x] new code has new specs
- [x] for new models - new factories were added too
- [x] unit specs are passed
- [x] code is well-formatted, Ruby, CSS, etc
- [x] screenshots of new pages, features are added
- [ ] code passed internal team code review
- [x] no instance variables in partials
- [x] code is well formatted and has spaces!!!

PS: delete your branch in github after merge http://prntscr.com/q80mvq

### CODE REVIEW

**Comment by vikdotdev** [comment]
2019-12-26T16:06:05
2019-12-26T16:06:24
2019-12-26T16:06:37

File: config/environments/production.rb [hunk 1]
```diff
@@ -16,11 +16,11 @@
 
   # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
   # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
-  # config.require_master_key = true
+  config.require_master_key = true
 
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
-  config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
+  config.public_file_server.enabled = true || ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (vikdotdev) [inline config/environments/production.rb]: тут напевно одне з двох: або просто залишити `true` або ж поставити env змінну на сервері - обидва зразу сенсу напевно немає?

Comment (igorkasyanchuk) [reply config/environments/production.rb]: цього немає бути
має бути як буор

Comment (romaslmd) [reply config/environments/production.rb]: є цікавіший прикол

`ENV.fetch('RAILS_SERVE_STATIC_FILES', true)`

File: config/initializers/any_login.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 AnyLogin.setup do |config|
   config.name_method = proc { |e| [[e.email, e.role].join(' || '), e.id] }
+  config.enabled = true
```
Comment (romaslmd) [inline config/initializers/any_login.rb]: disable on test env, there's no need in it there

File: config/initializers/any_login.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
 AnyLogin.setup do |config|
   config.name_method = proc { |e| [[e.email, e.role].join(' || '), e.id] }
-  config.enabled = true
+  config.enabled = true unless Rails.env == 'test'
```
Comment (romaslmd) [inline config/initializers/any_login.rb]: fyi: `config.enabled = Rails.env != 'test'`

**Pull request merged**